projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -42,6 +42,20 @@ public function reregister_menu_items() {
 			if ( $nudge ) {
 				parent::add_upsell_nudge( $nudge );
 			}
+
+			/**
+			 * Whether to show the WordPress.com Inbox menu under Upgrades menu.
+			 *
+			 * @use add_filter( 'jetpack_show_wpcom_inbox_menu', '__return_true' );
+			 * @module masterbar
+			 *
+			 * @since 9.7.0
+			 *
+			 * @param bool $jetpack_show_wpcom_inbox_menu Load the WordPress.com Inbox menu item. Default to false.
+			 */
+			if ( apply_filters( 'jetpack_show_wpcom_inbox_menu', false ) ) {
```

Comment:
For this to actually work in Calypso, this call needs to go above the `if ( ! $this->is_api_request ) {` call in line 37.